draft-nygren-httpbis-httpssvc.md

```diff
@@ -562,6 +517,39 @@ Authoritative DNS servers SHOULD return A, AAAA, and HTTPSSVC records
 (as well as any relevant CNAME records) in the Additional Section for
 any in-bailiwick SvcDomainNames.
 
+# Performance optimizations
+
+For optimal performance (i.e. minimum connection setup time), clients
+SHOULD issue address (AAAA and/or A) and HTTPSSVC queries
+simultaneously, and SHOULD implement a client-side DNS cache.
+With these optimizations in place, and conforming DNS servers,
+using HTTPSSVC does not add network latency to connection setup.
+
+A nonconforming recursive resolver might return an HTTPSSVC response with
+a nonempty SvcDomainName, without the corresponding address records.  If
+all the HTTPSSVC RRs in the response have nonempty SvcDomainName values,
+and the client does not have address records for any of these values in
+its DNS cache, the client SHOULD perform an additional address query for
+the selected SvcDomainName.
+
+The additional DNS query in this case introduces a delay.  To avoid
+causing a delay for clients using a nonconforming recursive resolver,
+domain owners SHOULD choose the SvcDomainName to be a name in the
+origin hostname's CNAME chain if possible.  This will ensure that the required
+address records are already present in the client's DNS cache.
+
+Highly performance-sensitive clients MAY implement the following special-
+case shortcut to avoid increased connection time: if (1) all of the
+HTTPSSVC records returned have SvcRecordType = 0, (2) none of their
+SvcDomainNames are in the DNS cache, and (3) the address queries for the
+origin domain return usable IP addresses, then the client MAY ignore the
+HTTPSSVC records and connect directly to the origin domain.
+
```

Comment:
Add:  "When the SvcDomainNames and any needed HTTPSSVC records are available, these client SHOULD make subsequent requests over connections specified by the HTTPSSVC records."

(I suspect that for transitional purposes the A/AAAA record alongside the origin domains in these cases will provide sub-par performance relative to the target of the HTTPSSVC record.)

Reply:
Moved to #19.